### PR TITLE
Closes issue #55

### PR DESCRIPTION
**Pull request by danirus**
Allows the user to save the sidebar settings in a `conf.cfg` file inside the `<conf_dir>`.

Provides a new class `Settings` that is used as an attribute of the `PythonInput` class. All the user customisable settings are provided in a dictionary in the constructor of the class `Settings`. Changing the settings has the same effect as in the current code. 

The `Settings` class provides two utility methods:
- `update_from`: that loads customisable settings from the `~/.ptpython/conf.cfg` (or another given path).
- `save_config`: that saves the settings in the `~/.ptpython/conf.cfg` file (or another given path).

I added a `done` parameter to the `embed`. The idea is to have a way to do something upon client termination. The utility of the `done` hook in this PR is to be able to save the settings to the file.

### CODE REVIEW

File: ptpython/config.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+option_re = re.compile('^(?P<option>\w+)\s+=.+(?P<comment>#.*)?$')
+
+
+class Settings:
+    """
+    Ptpython user defined settings, loaded from <config-dir>/conf.cfg file.
+
+    :param defaults: dictionary with repl's settings customisable in conf.cfg.
+    """
+    user_defined = {}
```
Comment (jonathanslenders) [inline ptpython/config.py]: This is not an instance variable. A second instance of ``Settings`` will share the same ``user_defined`` attribute.

File: ptpython/python_input.py [hunk 2]
```diff
@@ -153,40 +154,42 @@ def __init__(self,
...
-        self.show_sidebar = False  # Currently show the sidebar.
-        self.show_sidebar_help = True # When the sidebar is visible, also show the help text.
-        self.show_exit_confirmation = False  # Currently show 'Do you really want to exit?'
-        self.terminal_title = None  # The title to be displayed in the terminal. (None or string.)
-        self.exit_message = 'Do you really want to exit?'
-
-        # Tokens to be shown at the prompt.
-        self.prompt_style = 'classic'  # The currently active style.
+        self.settings = Settings({
+            'show_signature': True,
```
Comment (jonathanslenders) [inline ptpython/python_input.py]: I guess this could break many existing config files. I have to think about this.

File: ptpython/config.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        # so that they can be updated with new user defined values.
+        current_lines = []
+
+        settings = copy.copy(self.user_defined)
+
+        if os.path.exists(file_path):
+            with open(file_path, 'r') as f:
+                current_lines = f.readlines()
+
+        cfg_file = open(file_path, 'w')
```
Comment (jonathanslenders) [inline ptpython/config.py]: Without looking further into the code, this should either use a with-block or a try/finally.

File: ptpython/python_input.py (hunk 2 above)
Comment (danirus) [inline ptpython/python_input.py]: I guess so. I think at some point you can't prevent that from happening. In this case I think user customisable settings inside the class PythonInput have to distinguish themselves from method attributes. They have different purposes and behave differently too. Maybe doing the PR at the time of a major version change would be more convenient.

**Comment by danirus** [comment]
Put this on hold @jonathanslenders. I'm getting an unexpected behaviour when I launch ptpython with my changes. I'll let you know.

**Comment by jonathanslenders** [comment]
Don't worry, Daniel. It's not urgent.
See when you have time, I do appreciate the effort.

File: ptpython/python_input.py (hunk 2 above)
Comment (danirus) [inline ptpython/python_input.py]: A better approach could allow settings to be made in the repl object (repl.show_sidebar, etc, as in old config.py modules) to do not break backwards compatibility, provide a __setattr__ in  PythonInput to move those to the settings attribute, and display a warning FutureWarning to notify the change in the next release.

**Pull request closed without merging**